tfhe/tests/backward_compatibility/shortint.rs

```diff
@@ -0,0 +1,81 @@
+/// built-in types into parameters suitable for the currently tested version.
+fn load_params(test_params: TestParameterSet) -> ShortintParameterSet {
+    ShortintParameterSet::new_pbs_param_set(PBSParameters::PBS(ClassicPBSParameters {
+        lwe_dimension: LweDimension(test_params.lwe_dimension),
+        glwe_dimension: GlweDimension(test_params.glwe_dimension),
+        polynomial_size: PolynomialSize(test_params.polynomial_size),
+        lwe_noise_distribution: DynamicDistribution::new_gaussian_from_std_dev(StandardDev(
+            test_params.lwe_noise_gaussian_stddev,
+        )),
+        glwe_noise_distribution: DynamicDistribution::new_gaussian_from_std_dev(StandardDev(
+            test_params.glwe_noise_gaussian_stddev,
+        )),
+        pbs_base_log: DecompositionBaseLog(test_params.pbs_base_log),
+        pbs_level: DecompositionLevelCount(test_params.pbs_level),
+        ks_base_log: DecompositionBaseLog(test_params.ks_base_log),
+        ks_level: DecompositionLevelCount(test_params.ks_level),
+        message_modulus: MessageModulus(test_params.message_modulus),
+        carry_modulus: CarryModulus(test_params.carry_modulus),
+        max_noise_level: MaxNoiseLevel::new(test_params.max_noise_level),
+        log2_p_fail: test_params.log2_p_fail,
+        ciphertext_modulus: CiphertextModulus::try_new(test_params.ciphertext_modulus).unwrap(),
+        encryption_key_choice: {
+            match &*test_params.encryption_key_choice {
+                "big" => EncryptionKeyChoice::Big,
+                "small" => EncryptionKeyChoice::Small,
+                _ => panic!("Invalid encryption key choice"),
+            }
+        },
+    }))
+}
+
+fn test_shortint_ciphertext(test: ShortintCiphertextTest, dir: &Path) {
+    let key_file = dir.join(&*test.key_filename);
+    let key = ClientKey::unversionize(load_versioned(key_file).unwrap()).unwrap();
+    let ct_file = dir.join(&*test.ct_filename);
+    let ct = Ciphertext::unversionize(load_versioned(ct_file).unwrap()).unwrap();
+
+    let clear = key.decrypt(&ct);
+    assert_eq!(clear, test.clear_value);
+}
+
+fn test_shortint_clientkey(test: ShortintClientKeyTest, dir: &Path) {
+    let versioned_key = load_versioned(dir.join(&*test.key_filename)).unwrap();
+
+    let test_params = load_params(test.parameters);
+
+    let key = ClientKey::unversionize(versioned_key).unwrap();
+
+    assert_eq!(test_params, key.parameters);
+}
+
+pub fn test_shortint<P: AsRef<Path>>(base_dir: P) {
+    let shortint_dir = base_dir.as_ref().join("shortint");
+
+    let meta = load_metadata(shortint_dir.join("shortint.ron")).unwrap();
+
+    for testcase in meta {
+        match testcase {
+            TestMetadata::ShortintCiphertext(test) => test_shortint_ciphertext(test, &shortint_dir),
+            TestMetadata::ShortintClientKey(test) => test_shortint_clientkey(test, &shortint_dir),
+        }
+    }
```

Comment:
done: tests now return a result. Result are collected for all tests and dumped at the end. I didn't add `--nocapture` since the output is dumped in case of error, but I can change this if needed.

Reply:
the no capture is more for local development when running the tests to get results as the tests are going, this is likely fine as a first iteration

Reply:
Done, now test results are printed as they arrive